diff --git a/addon/error.js b/addon/error.js
index 4e2c0a28ae80c4d1821be9bb4063665201013078..d1b2620a6adf1bbb189683384247a63ef2aacb71 100644
--- a/addon/error.js
+++ b/addon/error.js
@@ -2,7 +2,7 @@
   @module @ember-data/adapter/error
  */
 import { assert } from '@ember/debug';
-import EmberError from '@ember/error';
+const EmberError = Error;
 
 /**
   A `AdapterError` is used by an adapter to signal that an error occurred
diff --git a/addon/rest.ts b/addon/rest.ts
index dbf44eab1e0b4e51e67b4b6cacee136c6107f4c2..f0f65a98b1f1a65f8ba168626047921596837aa9 100644
--- a/addon/rest.ts
+++ b/addon/rest.ts
@@ -4,7 +4,7 @@
 import { getOwner } from '@ember/application';
 import { deprecate, warn } from '@ember/debug';
 import { computed } from '@ember/object';
-import { assign } from '@ember/polyfills';
+const assign = Object.assign;
 import { join } from '@ember/runloop';
 import { DEBUG } from '@glimmer/env';
 
@@ -1303,8 +1303,8 @@ class RESTAdapter extends Adapter.extend(BuildURLMixin) {
 
   /**
     Used by `findAll` and `findRecord` to build the query's `data` hash
-    supplied to the ajax method. 
-   
+    supplied to the ajax method.
+
     @method buildQuery
     @since 2.5.0
     @public
